src/main/java/org/mule/tools/api/classloader/model/Artifact.java

```diff
@@ -34,18 +32,28 @@ public URI getUri() {
     return uri;
   }
 
+  public void setUri(URI uri) {
+    checkNotNull(uri, "Uri cannot be null");
+    this.uri = uri;
+  }
+
+  private void setArtifactCoordinates(ArtifactCoordinates artifactCoordinates) {
+    checkNotNull(artifactCoordinates, "Artifact coordinates cannot be null");
+    this.artifactCoordinates = artifactCoordinates;
+  }
+
   @Override
   public String toString() {
     return artifactCoordinates.toString();
   }
 
   @Override
   public int compareTo(Object that) {
-    return getMavenCoordinates(this).compareTo(getMavenCoordinates((Dependency) that));
+    return getMavenCoordinates(this).compareTo(getMavenCoordinates((Artifact) that));
   }
 
-  private String getMavenCoordinates(Dependency dependency) {
-    ArtifactCoordinates coordinates = dependency.getArtifactCoordinates();
+  private String getMavenCoordinates(Artifact artifact) {
```

Comment:
Made it protected and changed the implementation to use this.getArtifactCoordinates